.circleci/config.yml

```diff
@@ -1,99 +1,132 @@
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ arch }}-{{ checksum "yarn.lock" }}
+            - v1-dependencies-{{ arch }}
+      - run: npx yarn install --frozen-lockfile --ignore-engines
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ arch }}-{{ checksum "yarn.lock" }}
+  test:
+    steps:
+      - run: npx yarn run lint
+      - run:
+          name: Tests with code coverage
+          command: npx yarn run coverage
+          environment:
+            DEBUG: electron-rebuild
+      - run: npx yarn run codecov
 
-version: 2.1
-orbs:
-  win: circleci/windows@1.0.0
 jobs:
-  test-linux-12:
+  test-linux:
     docker:
-      - image: cimg/node:12.22
-    <<: *steps-test
-  test-linux-14:
-    docker:
-      - image: cimg/node:14.20
-    <<: *steps-test
-  test-linux-16:
-    docker:
-      - image: cimg/node:16.18
-    <<: *steps-test
+      - image: cimg/base:stable
+    parameters:
+      node-version:
+        description: Node.js version to install
+        type: string
+    steps:
+      - install:
+          node-version: << parameters.node-version >>
+      - test
   test-mac:
     macos:
       xcode: "13.3.0"
-    <<: *steps-test
+    resource_class: macos.x86.medium.gen2
+    parameters:
+      node-version:
+        description: Node.js version to install
+        type: string
+    steps:
+      - install:
+          node-version: << parameters.node-version >>
+      - test
   test-windows:
     executor:
-      name: win/vs2019
+      name: win/server-2022
       shell: bash.exe
     environment:
-      GYP_MSVS_VERSION: '2019'
-    <<: *steps-test
-
-  release:
-    docker:
-      - image: cimg/node:14.17
+      GYP_MSVS_VERSION: '2022'
+    parameters:
+      node-version:
+        description: Node.js version to install
+        type: string
     steps:
-      - checkout
-      - *step-restore-cache
-      - run: yarn install --frozen-lockfile --ignore-engines
-      - run: npx @continuous-auth/circleci-oidc-github-auth@1.0.4
-      - run: npx semantic-release@17.4.5
+      - run:
+          command: New-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem" -Name "LongPathsEnabled" -Value 1 -PropertyType DWORD -Force
+          shell: powershell.exe
+      - install:
+          node-version: << parameters.node-version >>
+      - test
 
 workflows:
-  version: 2
   test_and_release:
     # Run the test jobs first, then the release only when all the test jobs are successful
     jobs:
-      - test-linux-16
-      - test-linux-12
-      - test-linux-14
-      - test-mac
-      - test-windows
-      - release:
+      - test-linux:
+          matrix:
+            parameters:
+              node-version:
+                - 20.2.0
+                - 18.16.0
+                - 16.20.0
+                - 14.21.3
+                - 12.22.12
```

Comment:
Why continue to support and encourage the use of end-of-life versions of Node.js?  If users want support then they can try to reproduce their problems on a supported version.  If folks are paying for long-term support for an OS that has an EOL version of Node.js then their OS provider can support them. If folks are using an OS that is beyond the standard support and are not paying for LTS, then their problems should be demonstrated on supported versions.  Building on secure platforms and being able to use new language optimizations and features helps everyone's safety and velocity so let's encourage security and performance upgrades.

Reply:
This package has an engine version that supports Node.js v12 and above, so dropping support would be a breaking change (and require a major version bump on this package). Any major version bump causes fragmentation with the user base, as can be seen [in the npm downloads stats for older versions of this package](https://www.npmjs.com/package/electron-rebuild?activeTab=versions) (it used to be called just `electron-rebuild`) where there are still ~6k downloads/week on the final 2.y.z version and ~7.5k downloads/week on the final 1.y.z version. We want to avoid furthering that fragmentation, so we don't want to bump majors more often than we need to (which every Node.js EOL would require). These packages also don't have a ton of maintainer time to put into them, so it takes some time for things to be merged - if we bump the major version we might make the safety issue worse by putting new fixes behind a new major version where they won't be picked up by some subset of the user base. For ease of maintenance these packages aren't setup to do additional releases to old majors, so once we bump majors there are no more fixes being backported to old majors (and we would need the CI to be testing the old Node.js versions those majors support to be able to do so).

So the general policy of @~electron/wg-ecosystem is to continue to test and not explicitly drop older Node.js versions unless necessary. We're not encouraging the usage of EOL versions of Node.js, more avoiding breaking them when feasible. In most cases we won't go out of our way to fix bugs which only affect those EOL versions, and if a future change requires dropping support for older EOL versions of Node.js we will explicitly drop support then, if it's deemed necessary.

Hope this makes sense. 🙂

Reply:
Users of legacy software create a lot of noise in support channels...
* 123 issues: https://github.com/search?q=repo%3Anodejs%2Fnode-gyp+electron-rebuild&type=issues

Reply:
Will that noise really go away if this package stops testing those EOL Node.js versions, though? Even if we drop support and bump the major, some users will stick on the current major (as seen in the npm download stats listed above).

If you tweak the GitHub search to include `created:>2022-01-01`, it only returns 16 issues in the past 18 months, and without going beyond flipping through the descriptions shown on the search page, at least 3 of them were reporting supported Node.js versions.

It's also worth noting that `node-gyp` itself is still showing Node.js v12 and v14 as supported engines according to `package.json`: https://github.com/nodejs/node-gyp/blob/33391db3a0008eff8408890da6ab232f2f90fcab/package.json#L37-L39

Reply:
> Will that noise really go away if this package stops testing those EOL Node.js versions?

Rephrase: Will that noise **decrease** if this package stops **encouraging / supporting** EOL Node.js versions?

Yes, it will.  Developers are improving software and ignoring their efforts is suboptimal on many different levels.  Solutions that used to work in older versions of npm, node, and gyp no longer work today.  Supporting both legacy and current is complex, frustrating, and error-prone.  Doing so for the sake of statistics is a shame.